tools/azure-sdk-tools/packaging_tools/auto_codegen.py

```diff
@@ -37,6 +40,54 @@ def init_new_service(package_name, folder_name):
             with open(str(ci), 'w') as file_out:
                 file_out.writelines(content)
 
+def update_service_metadata(sdk_folder, data, config, folder_name, package_name, spec_folder, input_readme):
```

Comment:
Please add some unit tests for this method with different test cases:
- Create metadata
- Update metadata
- Update MANIFETS.IN
- No need to update Update MANIFETS.IN
- Exception thrown from `update_service_metadata`